### DISCUSSION TITLE
Advantages over Elasticsearch kNN

### DISCUSSION THREAD

**Discussion by andrasaponyi** [alexklibisz/elastiknn#511] [Q&A] [answered]
Hi,

I'm learning about vector similarity search in Elasticsearch and have recently come across this plugin as an alternative to the default kNN implementation. I understand that previously Elasticsearch did not support approximate nearest-neighbor search, [but aNN was actually  implemented in version 8.0](https://www.elastic.co/blog/introducing-approximate-nearest-neighbor-search-in-elasticsearch-8-0).

So I've been wondering - what are the advantages of elastiknn over the native Elasticsearch implementation at this point?

Here are some advantages I can see from what I've read so far:

- elastiknn supports multiple vectors per document
- some similarity metrics not supported by ES (L1, Jaccard, Hamming)
- elastiknn uses Locality Sensitive Hashing as opposed to HNSW (but is this really an advantage?)

Looking forward to your answers 🙂

**Comment by alexklibisz** [comment]
Hi, I have only used Elasticsearch kNN for some very quick/simple experiments, several months ago. So take this with that grain of salt.

The first two you pointed out are accurate. Elastiknn also supports sparse vectors; AFAIK, elasticsearch does not.

Whether LSH is an advantage I guess depends on the problem and dataset at hand. I think HNSW is better on most benchmarks.

**Reply by andrasaponyi** [reply, marked as answer]
Thank you, @alexklibisz, that's very helpful!
